extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-workspace-service-impl.ts

```diff
@@ -71,8 +71,17 @@ export class K8sWorkspaceServiceImpl implements WorkspaceService {
   }
 
   public async getWorkspaceSettings(): Promise<WorkspaceSettings> {
-    console.log('workspaceService.getWorkspaceSettings() not supported');
-    return {};
+    const publicURI = this.env.getPluginRegistryURL();
+    const privateURI = this.env.getPluginRegistryInternalURL() || publicURI;
```

Comment:
Is there a real reason for using a public URI as a private one if it is not set?
Considering we're already checking that in https://github.com/eclipse-che/che-theia/blob/f677d115a50ab18cc4ae3ac25ce7236759d2f907/extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts#L103

Reply:
I'm not asking about both URIs.
I'm asking about both places of using it:
1. https://github.com/eclipse-che/che-theia/blob/4f6995d3ab8aad15669845e2de5b1464f79f8933/extensions/eclipse-che-theia-remote-impl-k8s/src/node/k8s-workspace-service-impl.ts#L75
2. https://github.com/eclipse-che/che-theia/blob/f677d115a50ab18cc4ae3ac25ce7236759d2f907/extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts#L103

In other words, if I want to get and see the "raw" settings returned by `workspaceService.getWorkspaceSettings()`, I should be able to get it as it is. Without any substitutions inside, like in 1.
And only when I get the "raw" settings, I can do with them whatever I want - like making such checks as in 2.

Reply:
> Private URI is used to get the plugin list on the backend, public URI is used by frontend to load plugin icons

`workspaceService.getWorkspaceSettings()` knows nothing about that ^. And it **must not** know that.
It's only `ChePluginService` who has this knowledge about plug-ins, icons and so on.

Reply:
Let's imagine, in some other place in the code, I call `workspaceService.getWorkspaceSettings()`
in order to see if `privateURI` is absent. Despite the presence of `publicURI`.
How I can know that? There's no chance.
That's why I'm proposing to return the `WorkspaceSettings` as it is. And keep such checking only in the place where it's actually important - like in `ChePluginService`.